components/skus/browser/rs/lib/src/sdk/credentials/fetch.rs

```diff
@@ -63,17 +94,123 @@ where
             return Err(InternalError::OrderUnpaid.into());
         }
 
-        let mut csprng = OsRng;
-
         for item in order.items {
             match item.credential_type {
+                CredentialType::TimeLimitedV2 => {
+                    let mut num_creds: usize = 0;
+                    if let Some(ref metadata) = order.metadata {
+                        let num_intervals = metadata.num_intervals.unwrap_or(0);
+                        let num_per_interval = metadata.num_per_interval.unwrap_or(0);
+                        if num_intervals == 0 && num_per_interval == 0 {
+                            return Err(InternalError::OrderMisconfiguration.into());
+                        }
+                        num_creds = num_intervals * num_per_interval;
+                    }
+
+                    let blinded_creds: Vec<BlindedToken> =
+                        match self.client.get_time_limited_v2_creds(&item.id).await? {
+                            Some(item_creds) => {
+                                match item_creds.state {
+                                    CredentialState::GeneratedCredentials => {
+                                        event!(Level::DEBUG, "generated credentials state");
+                                    }
+                                    CredentialState::SubmittedCredentials => {
+                                        event!(Level::DEBUG, "submitted credentials state");
+                                    }
+                                    CredentialState::FetchedCredentials => {
+                                        event!(Level::DEBUG, "fetched credentials state");
+                                    }
+                                    CredentialState::ActiveCredentials => {
+                                        event!(Level::DEBUG, "active credentials state");
+                                    }
+                                };
```

Comment:
can we replace this whole thing with something like:
```
event!(Level::DEBUG, state = item_creds.state);
```

Reply:
dont actually need this anymore anyways.